packages/angular/cli/src/command-builder/utilities/json-schema.ts

```diff
@@ -43,6 +43,62 @@ export interface Option extends yargs.Options {
    * If this is falsey, do not report this option.
    */
   userAnalytics?: string;
+
+  /**
+   * Type of the values in a key/value pair field.
+   */
+  itemValueType?: 'string';
+}
+
+function coerceToStringMap(
+  dashedName: string,
+  value: (string | undefined)[],
+): Record<string, string> | Promise<Record<string, string>> {
+  const stringMap: Record<string, string> = {};
+  for (const pair of value) {
+    // This happens when the flag isn't passed at all.
+    if (pair === undefined) {
+      continue;
+    }
+
+    const eqIdx = pair.indexOf('=');
+    if (eqIdx === -1) {
+      // TODO: Remove workaround once yargs properly handles thrown errors from coerce.
+      // Right now these sometimes end up as uncaught exceptions instead of proper validation
+      // errors with usage output.
+      return Promise.reject(
+        new Error(
+          `Invalid value for argument: ${dashedName}, Given: '${pair}', Expected key=value pair`,
+        ),
+      );
+    }
+    const key = pair.slice(0, eqIdx);
+    const value = pair.slice(eqIdx + 1);
+    stringMap[key] = value;
+  }
+
+  return stringMap;
+}
+
+function isStringMap(node: json.JsonObject) {
```

Comment:
```suggestion
function isStringMap(node: json.JsonObject): boolean {
```

Reply:
Might also want to consider combining the multiple logical statements.

Reply:
Grouped into "exclude other kinds of objects" and "define desired kind of objects".